common/lcs/src/error.rs

```diff
@@ -13,8 +13,10 @@ pub enum Error {
     Eof,
     #[error("I/O error: {0}")]
     Io(String),
-    #[error("exceeded max sequence length")]
+    #[error("exceeded max sequence length: {0}")]
     ExceededMaxLen(usize),
+    #[error("exceeded max object depth while entering: {0}")]
+    ExceededObjectDepthLimit(&'static str),
     #[error("expected boolean")]
```

Comment:
What I'm getting at is this is a recursion limit (i.e. an implementation constraint) rather than a statement on a predefined notion of structural depth that would exist in Rust independently of LCS  / Serde details.
How about "recursion depth" wherever you use "object depth", especially in the initial doc comment of lib.rs?